src/uu/stat/src/stat.rs

```diff
@@ -144,24 +175,29 @@ trait ScanUtil {
 }
 
 impl ScanUtil for str {
+    /// Scans for a number at the beginning of the string
+    /// Returns the parsed number and the character count
+    /// Since we only deal with ASCII characters (+, -, 0-9), character count equals byte count
     fn scan_num<F>(&self) -> Option<(F, usize)>
     where
         F: std::str::FromStr,
     {
         let mut chars = self.chars();
-        let mut i = 0;
+        let mut count = 0;
+
         match chars.next() {
-            Some('-' | '+' | '0'..='9') => i += 1,
+            Some('-' | '+' | '0'..='9') => {
+                count += 1;
+            }
             _ => return None,
         }
-        for c in chars {
-            match c {
-                '0'..='9' => i += 1,
-                _ => break,
-            }
+
+        for _ in chars.take_while(char::is_ascii_digit) {
+            count += 1;
         }
-        if i > 0 {
-            F::from_str(&self[..i]).ok().map(|x| (x, i))
```

Comment:
I think you can simplify it with something like:
```rust
let count = chars
    .next()
    .filter(|&c| c.is_ascii_digit() || c == '-' || c == '+')
    .map(|_| 1 + chars.take_while(char::is_ascii_digit).count())
    .unwrap_or(0);
```

Reply:
impressive! thanks